IPython/terminal/prompts.py

```diff
@@ -19,7 +26,7 @@ def in_prompt_tokens(self, cli=None):
 
     def _width(self):
         in_tokens = self.in_prompt_tokens()
-        return sum(len(s) for (t, s) in in_tokens)
+        return sum(len(s) for (t, s) in in_tokens if t is not ZeroWidthEscape)
```

Comment:
Please, just use ``prompt_toolkit.layout.utils.token_list_width``: https://github.com/jonathanslenders/python-prompt-toolkit/blob/master/prompt_toolkit/layout/utils.py#L27

Unless there are objections, ``ZeroWidthEscape`` along with several bug fixes, and a few small improvements, is probably already going in the next release: 1.0.1.  (Backwards-compatibility is very important, but right now, I'm not sure, whether each new feature needs in y-increment in x.y.z. In the future definitely, yes.)